Bugs in GANs

harbac opened the pull request.

harbac committed:
Update cyclegan.py

Removed pool layers

harbac committed:
Update cyclegan.py

Instance norm should not be there in the last layer

harbac committed:
Update dcgan.py

Spelling mistakes!!! (Because of which activations were not being applied properly)

harbac committed:
Update dcgan.py

Changed the dimensions of the hidden channels

harbac committed:
Update dcgan.py

Embedding layer missing from weight initialization.

JesusPinedaC commented:
Hi @HarshithBachimanchi 

I've reviewed the changes and I think they look good, especially if the examples work as expected now. 

However, I suggest that we implement unit tests for both CycleGAN and DCGAN. This will help us keep track of any possible changes to the models. The unit tests can be quite simple and focus on evaluating the correct order of operations, expected number of layers, and appropriate output dimensions of the model.

harbac committed:
Delete generator.py

harbac committed:
Update cyclegan.py

BenjaminMidtvedt committed:
Allow to set tensors parameter to control what is initialized

BenjaminMidtvedt committed:
Merge branch 'bm/weight-initialization' into pr/HarshithBachimanchi/103

harbac committed:
Update dcgan.py

Normalization is applied only to the weights and not biases

harbac committed:
Update dcgan.py

Normalization is now applied only to the weights and not biases. Removed nn.Linear layer from the normalization as it does not exist in dcgan generator.

harbac committed:
Update dcgan.py

harbac committed:
Update cyclegan.py

harbac committed:
Update dcgan.py

harbac committed:
Create test_cyclegan.py

harbac committed:
Update test_dcgan.py

harbac committed:
Create __init__.py

harbac committed:
Update test_cyclegan.py

harbac committed:
Update test_dcgan.py

harbac commented:
Hi! @JesusPinedaC 

I have created the unit tests now, and adjusted few other minor things. The models are working as expected now, and the weights are being initialized properly for dcgan.

If everything looks good, I think this can be merged now.

BenjaminMidtvedt commented:
These are excellent unittests! @HarshithBachimanchi 

Everything looks good to me.
